### ISSUE TITLE
Pooler output - Question

### ISSUE THREAD

**Issue by anindabitm** [agemagician/ProtTrans#24]
Hi,

Many thanks for providing pre-trained models. I was trying to use pre-trained model ProtBert for Protein Sequence Classification task. You have also provided examples and have shown two alternatives:

1) Using feature extraction and then using features to train a new model or use features as inputs to Random Forest, XGBoost for classification. Understand this would need careful model selection to achieve good results

2) In the second alternative, you have suggested full model fine tuning by adding a classification head. 

In both the above cases, I noticed you have used the last hidden state output of the Bert Model for the downstream task. Since the Bert Model also provides a pooler output, I could not find any use of that in your examples. Am I missing something ?

If I am using Pooler output, I am not getting very good results. Could you please provide some inputs.

`TOKENIZER = AutoTokenizer.from_pretrained("Rostlab/prot_bert", do_lower_case=False,use_fast=True)
MODEL = AutoModel.from_pretrained("Rostlab/prot_bert")
MAX_LEN = 256
class ProtBert(nn.Module):
    def __init__(self):
        super(ProtBert, self).__init__()
        self.bert = MODEL
        self.bert_drop = nn.Dropout(0.3)
        self.out = nn.Linear(1024, 20)
    
    def forward(self,id,mask,token):
      id = id
      mask = mask
      token = token
      o2 = self.bert(
            input_ids = id,
            attention_mask = mask,
            token_type_ids = token
      )
      bo = self.bert_drop(o2['pooler_output'])
      output = self.out(bo)
      return output

`

**Comment by agemagician** [comment]
Hi @anindabitm ,

The original Bert version uses the pooler for next sentence prediction during training, which we didn't use, because in protein sequences we don't have this concept.
Each protein sequence is like a complete document in NLP.

That's why you should use something like mean pooling over last hidden state output rather than the pooler.
